geomstats/_backend/pytorch/__init__.py

```diff
@@ -125,10 +124,21 @@ def logical_and(x, y):
 
 
 def any(x, axis=None):
-    if axis is None and torch.is_tensor(x):
+    if not torch.is_tensor(x):
+        x = torch.tensor(x)
+    if axis is None:
         return x.bool().any()
-    numpy_result = _np.array(_np.any(_np.array(x), axis=axis))
-    return torch.from_numpy(numpy_result)
+    if isinstance(axis, int):
+        return torch.any(x.bool(), axis)
+    if len(axis) == 2:
+        axis = list(axis)
+        for i_axis, one_axis in enumerate(axis):
+            if one_axis < 0:
+                axis[i_axis] = ndim(x) + one_axis
+        return torch.any(
+            torch.any(x.bool(), axis[1]), axis[0])
+    raise NotImplementedError(
+        'any not implemented for more than two axes.')
```

Comment:
what about recursively calling any, like:
```python

def any(x, axis=None):
    if axis is None:
        return torch.any(x)
    if isinstance(axis, int):
        return torch.any(x, axis)
    if len(axis) == 1:
        return torch.any(x, *axis)
    else:
        new_axis = tuple(k - 1 if k >= 0 else k for k in axis[1:])
        print(new_axis)
        return any(torch.any(x, axis[0]), new_axis)
```
Just something to think about, the above doesn't work if the axes are not ordered...

Reply:
Thanks for spotting this! I have added the recursive logic.

Reply:
Also added it for `all`.